src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -0,0 +1,230 @@
+import time
+import colorama
+from azure.common import AzureHttpError
+from azure.storage.blob import AppendBlobService, ContainerPermissions
+from .sdk.models import (
+    QueueBuildRequest,
+    DockerBuildParameters,
+    PlatformProperties,
+    Build,
+    BuildArgument
+)
+from azure.cli.core.commands import LongRunningOperation
+import pytz
+
+
+def acr_show_logs(cmd, client, registry_name, build_id, resource_group_name=None):
+    # Required and hardcoded for now. Refer to the "HACK" introduced in build_operations.py
+    custom_headers = dict()
+    custom_headers['logType'] = 'RawText'
+    buildLogResult = client.get_log_link(
+        build_id=build_id, resource_group_name=resource_group_name, registry_name=registry_name, custom_headers=custom_headers)
+    logFileSas = buildLogResult.log_link
+
+    if not logFileSas:
+        return 'No logs found.'
+
+    match = get_match(logFileSas)
+    accountName = match.group('accountName')
+    containerName = match.group('containerName')
+    blobName = match.group('blobName')
+    sasToken = match.group('sasToken')
+
+    if not accountName:
+        raise ValueError(
+            "Unable to parse the account name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not containerName:
+        raise ValueError(
+            "Unable to parse the container name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not blobName:
+        raise ValueError(
+            "Unable to parse the blob name from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    if not sasToken:
+        raise ValueError(
+            "Unable to parse the SAS token from the log URL: '{!s}'."
+            .format(logFileSas)
+        )
+
+    blobService = AppendBlobService(account_name=accountName,
+                                    sas_token=sasToken)
+
+    byteSize = 1024*4
+    timeoutInMinutes = 10
+    timeoutInSeconds = timeoutInMinutes * 60
+    now = time.time()
+    stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName)
+
+
+def get_match(sasURL):
+    return re.search(
+        (r"http(s)?://(?P<accountName>.*?)\..*?/(?P<containerName>.*?)/"
+            r"(?P<blobName>.*?)\?(?P<sasToken>.*)"), sasURL)
+
+
+def stream_logs(byteSize, timeoutInSeconds, blobService,
+                containerName, blobName):
+
+    colorama.init()
+    stream = BytesIO()
+    metadata = dict()
+    start = 0
+    end = byteSize - 1
+    available = 0
+    lastModified = None
+
+    # Try to get the initial properties so there's no waiting.
+    # If the storage call fails, we'll just sleep and try again after.
+    try:
+        props = blobService.get_blob_properties(
+            container_name=containerName, blob_name=blobName)
+        metadata = props.metadata
+        available = props.properties.content_length
+        lastModified = props.properties.last_modified
+    except:
+        pass
```

Comment:
Can this try block be moved into the following while loop since it will be retried and the same code appeared again?

Reply:
If a log file is fully uploaded and the client is reusing this method, we would print out "No additional logs found ..." instead of printing out the entire section if we did that. It needs to hit the inner loop before going to the outer in that case.